.github/workflows/ci.yml

```diff
@@ -51,24 +76,11 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-cargo-build-target-
 
-      - name: Fmt
-        uses: actions-rs/cargo@v1
-        with:
-          toolchain: ${{ env.nightly }}
-          command: fmt
-
-      - name: Clippy
-        uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-features -- -A clippy::type_complexity -A clippy::identity_op -A clippy::boxed_local -D dead_code
-          toolchain: ${{ env.nightly }}
-
       - name: Build
```

Comment:
Contrary to my previous comment, here, since build and test steps are not helping each other, they could be separate jobs.

Reply:
I was confused by the cache step here and did not want to touch it. But yes it will probably be split

Reply:
I see your point. I think it would be good to have the unit test as a separate job since the build for tests is totally different from the normal build.